packages/gen-ai/bff/internal/integrations/kubernetes/token_k8s_client.go

```diff
@@ -1023,11 +1022,50 @@ server:
 // getModelDetailsFromServingRuntime queries the serving runtime and inference service
 // to get detailed model configuration information
 func (kc *TokenKubernetesClient) getModelDetailsFromServingRuntime(ctx context.Context, namespace string, modelID string) (map[string]interface{}, error) {
-	// Find InferenceService by name
+	// First try to find InferenceService by name
 	targetISVC, err := kc.findInferenceServiceByModelName(ctx, namespace, modelID)
 	if err != nil {
-		kc.Logger.Error("failed to find InferenceService for model", "modelID", modelID, "error", err)
-		return nil, fmt.Errorf("InferenceService for model '%s' not found: %w", modelID, err)
+		kc.Logger.Warn("unable to find InferenceService for model, trying LLMInferenceService", "modelID", modelID, "error", err)
+
+		// Fallback to LLMInferenceService
+		targetLLMSVC, llmErr := kc.findLLMInferenceServiceByModelName(ctx, namespace, modelID)
+		if llmErr != nil {
+			kc.Logger.Error("failed to find both InferenceService and LLMInferenceService for model", "modelID", modelID, "inferenceServiceError", err, "llmInferenceServiceError", llmErr)
+			return nil, fmt.Errorf("neither InferenceService nor LLMInferenceService for model '%s' found: InferenceService error: %w, LLMInferenceService error: %w", modelID, err, llmErr)
+		}
+
+		// Extract endpoint from LLMInferenceService
+		endpointURL, extractErr := kc.extractEndpointFromLLMInferenceService(targetLLMSVC)
+		if extractErr != nil {
+			kc.Logger.Error("failed to extract endpoint from LLMInferenceService", "modelID", modelID, "error", extractErr)
+			return nil, fmt.Errorf("failed to extract endpoint from LLMInferenceService for model '%s': %w", modelID, extractErr)
+		}
+
+		// Extract additional metadata from the LLMInferenceService
+		metadata := map[string]interface{}{}
+		if targetLLMSVC.Annotations != nil {
+			if displayName, exists := targetLLMSVC.Annotations[DisplayNameAnnotation]; exists {
+				metadata["display_name"] = displayName
+			}
+			if description, exists := targetLLMSVC.Annotations[InferenceServiceDescriptionAnnotation]; exists {
+				metadata["description"] = description
+			}
+		}
+
+		// All models are LLM models using vllm-inference
+		providerID := "vllm-inference"
```

Comment:
we need a constant here; and the value must match with the vllm provider's provider id created in the configmap